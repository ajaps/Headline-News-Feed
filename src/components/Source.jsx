import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import * as myActions from '../actions/Headlines';
/**
 * Represents Source template.
 */
export default class Source extends React.Component {
  constructor(props) {
    super(props);
    this.sourceItem = this.props.source;
    this.setSources = this.setSources.bind(this, this.sourceItem);
  }

/**
 * calls an action to set sources url and available sorting filter for articles
 * @param {Object} e The calling object property
 * @return {void}
 */
  setSources(sourceData) {
    const articleData = { id: sourceData.id,
      sortBysAvailable: sourceData.sortBysAvailable };
    myActions.fetchArticles(articleData);
  }

/**
 * @returns {component} A component with relevant source.
 */
  render() {
    const activeSource = this.props.activeSource;
    const btnClass = classNames({
      sourceBorder: activeSource === this.sourceItem.id,
    });
    return (
      <li className={btnClass}>
        <a onClick={this.setSources} ref="setSource">
          {this.sourceItem.name}
        </a>
      </li>
    );
  }
}

Source.propTypes = {
  activeSource: PropTypes.string,
  source: PropTypes.object,
};
