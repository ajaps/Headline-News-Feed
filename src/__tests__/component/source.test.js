import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import ReactTestUtils from 'react-dom/test-utils';
import Source from '../../components/Source.jsx';

jest.mock('../../api/NewsApi', () => ({
  fetchApiArticles: () => Promise.resolve('getPromise')
}));

describe('Sources component', () => {
  let app;
  const sourceSelected = 'bbc-sport';
  const sourceObject =
    {
      id: 'bbc-sport',
      name: 'BBC SPORT',
      sortBysAvailable: [
        'popular',
        'latest',
      ]
    };

  it('should render as expected', () => {
    app = shallow(
      <Source
          activeSource={sourceSelected}
          key={sourceObject.id}
          source={sourceObject}
      />
    );
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should call the set category function', () => {
    const renderedDoc = ReactTestUtils.renderIntoDocument(
      <Source
          activeSource={sourceSelected}
          key={sourceObject.id}
          source={sourceObject}
      />
    );
    const setSourceBtn = renderedDoc.refs.setSource;
    ReactTestUtils.Simulate.click(setSourceBtn);
  });
});
