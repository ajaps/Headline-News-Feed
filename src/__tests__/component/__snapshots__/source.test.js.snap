// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sources component should render as expected 1`] = `
<li
  className="sourceBorder"
>
  <a
    onClick={[Function]}
  >
    BBC SPORT
  </a>
</li>
`;
